File: roundC/PerfectSubarray.py
import sys
input = sys.stdin.readline

T = int(input())
for case in range(1, T + 1):
    N = int(input())
    arr = list(map(int, input().split()))
    max_val = 0
    for a in arr:
        max_val += max(a, 0)

    squares = set()
    i = 0
    while i ** 2 <= max_val:
        squares.add(i ** 2)
        i += 1
    cnt = 0
    for i in range(N):
        cum = 0
        for j in range(i, N):
            cum += arr[j]
            if cum in squares:
                cnt += 1
    print("Case #{0}: {1}".format(case, cnt))

File: roundC/PerfectSubarray.kt
import kotlin.math.*

fun main(args: Array<String>) {
    val T = readLine()!!.toInt()
    for (case in 1..T) {
        val N = readLine()!!.toInt()
        val list = readLine()!!.split(" ").map(String::toInt)
        var max_val = 0
        list.forEach{a ->
            max_val += max(a, 0)
        }
        var squares:MutableSet<Int> = mutableSetOf()
        var i = 0
        while(i * i <= max_val) {
            squares.add(i * i)
            i += 1
        }
        var cnt = 0
        for (i in 0..N-1) {
            var cum = 0
            list.slice(i..N-1).forEach{
                cum += it
                if (cum in squares) {
                    cnt += 1
                }
            }
        }
        println("Case #" + case.toString() + ": " + cnt.toString())
    }
} 